src/analyses/constant_propagator.h

```diff
@@ -175,14 +202,19 @@ class constant_propagator_ait:public ait<constant_propagator_domaint>
 
   constant_propagator_ait(
     goto_functionst::goto_functiont &goto_function,
-    const namespacet &ns):dirty(goto_function)
+    const namespacet &ns,
+    should_track_valuet should_track_value = track_all_values):
+    dirty(goto_function),
+    should_track_value(should_track_value)
   {
     operator()(goto_function, ns);
     replace(goto_function, ns);
   }
 
   dirtyt dirty;
 
+  std::function<bool(const exprt &, const namespacet &)> should_track_value;
```

Comment:
I'm not a fan of this being a public member. IMHO would be better to have this private. If this needs to be called from the outside, we could just call it `should_track_value_handler` and have a method `should_track_value` call this. It being public just invites shenanigans with people manually overwriting it.